justbuild.it-WebApp/src/test/java/justbuild/it/web/app/service/OfferSearchingServiceTest.java

```diff
@@ -0,0 +1,144 @@
+    private static final Offer OFFER_5 = new Offer(5L, ServiceCategoryEnum.ELECTRICITY, "Instalacje elektryczne", "Radom", new User("Tadeusz", "Wolski", "InstallWol", "install@example.com", "435657345"), LocalDateTime.now().minusDays(1));
+    private static final Offer OFFER_6 = new Offer(6L, ServiceCategoryEnum.INSTALLATION, "Kanalizacja", "Kraków", new User("Andrzej", "Redlin", "RedLine", "redline@example.com", "567897546"), LocalDateTime.now().minusDays(3));
+    private static final Offer OFFER_7 = new Offer(7L, ServiceCategoryEnum.GARDEN, "Aranżacja ogrodu", "Nowy Sącz", new User("Eustachy", "Konieczko", "Gardens", "gardens@example.com", "1232343212"), LocalDateTime.now().minusDays(6));
+
+    private static final List<Offer> GIVEN_TEST_LIST = List.of(OFFER_1, OFFER_2, OFFER_3, OFFER_4, OFFER_5, OFFER_6, OFFER_7);
+
+    @BeforeEach
+    public void setUpTest() {
+        offerFileRepositoryMock = mock(OfferFileRepository.class);
+        offerSearchingService = new OfferSearchingService(offerFileRepositoryMock);
+    }
+
+    @AfterEach
+    public void restoreSetUpTest() {
+        offerFileRepositoryMock = null;
+        offerSearchingService = null;
+    }
+
+    @Test
+    void getOffersList_shouldReturnOfferList() {
+        //Given
+        List<Offer> expectedOffersList = GIVEN_TEST_LIST;
+        when(offerFileRepositoryMock.getOffersFromJsonFile(anyString())).thenReturn(GIVEN_TEST_LIST);
+        // When
+        List<Offer> testOffersList = offerSearchingService.getOffersList();
+        // Then
+        assertThat(testOffersList)
+                .isNotNull()
+                .hasSize(expectedOffersList.size())
+                .containsExactlyElementsOf(expectedOffersList);
+    }
+
+    @Test
+    void getOffersList_shouldReturnEmptyList() {
+        //Given
+        when(offerFileRepositoryMock.getOffersFromJsonFile(anyString())).thenReturn(Collections.emptyList());
+        // When
+        List<Offer> testOffersList = offerSearchingService.getOffersList();
+        // Then
+        assertThat(testOffersList)
+                .isEmpty();
+    }
+
+    @ParameterizedTest
+    @MethodSource("getSearchingValueAndExpectedList")
+    void getOffersListFilteredBySearchValue_shouldReturnListFilteredBySearchValue(String searchingValue, List<Offer> expectedOffersList) {
+        //Given
+        when(offerFileRepositoryMock.getOffersFromJsonFile(anyString())).thenReturn(GIVEN_TEST_LIST);
+        //When
+        List<Offer> testOfferList = offerSearchingService.getOffersListFilteredBySearchValue(searchingValue);
+        //Then
+        assertThat(testOfferList)
+                .isNotNull()
+                .hasSize(expectedOffersList.size())
+                .containsExactlyElementsOf(expectedOffersList);
+    }
+
+    static Stream<Arguments> getSearchingValueAndExpectedList() {
```

Comment:
to na sam dół kodu, żeby czytelność była lepsza. Konwencja jest taka, że metody pomocnicze daje się zazwyczaj na dole.